lib/OpenQA/Script/CloneJob.pm

```diff
@@ -206,38 +206,50 @@ sub openqa_baseurl {
 }
 
 sub clone_job {
-    my ($jobid, $options, $clone_map, $depth) = @_;
-    $clone_map //= {};
-    $depth     //= 0;
+    my ($jobid, $options, $clone_map, $depth, $parent_jobid) = @_;
+    $clone_map    //= {};
+    $depth        //= 0;
+    $parent_jobid //= 0;
     return $clone_map->{$jobid} if defined $clone_map->{$jobid};
 
     my ($ua, $local, $local_url, $remote, $remote_url) = create_url_handler($options);
-    my $job = clone_job_get_job($jobid, $remote, $remote_url, $options);
-    if ($job->{parents}) {
-        my ($chained, $directly_chained, $parallel);
-        unless ($options->{'skip-deps'}) {
-            unless ($options->{'skip-chained-deps'}) {
-                $chained          = $job->{parents}->{Chained};
-                $directly_chained = $job->{parents}->{'Directly chained'};
+    my $job       = clone_job_get_job($jobid, $remote, $remote_url, $options);
+    my @job_types = 'parents';
+    push @job_types, 'children' if $options->{'clone-children'} && $depth == 0;
+    my $child_list;
+    for my $job_type (@job_types) {
+        if ($job->{$job_type}) {
+            my ($chained, $directly_chained, $parallel);
+            unless ($options->{'skip-deps'}) {
+                unless ($options->{'skip-chained-deps'}) {
+                    $chained          = $job->{$job_type}->{Chained};
+                    $directly_chained = $job->{$job_type}->{'Directly chained'};
+                }
+                $parallel = $job->{$job_type}->{Parallel};
+            }
+            $chained          //= [];
+            $directly_chained //= [];
+            $parallel         //= [];
+
+            print "Cloning dependencies of $job->{name}\n" if (@$chained || @$directly_chained || @$parallel);
+            for my $dependencies ($chained, $directly_chained, $parallel) {
+                # don't clone parallel jobs yet if children job type
+                if (@$dependencies && $dependencies == $parallel && $job_type eq 'children') {
+                    push @$child_list, @$parallel;    # keep the children's jobids
```

Comment:
Couldn't this be moved outside of the loop? From what I can understand, same results could be reached by doing:
```
push @$child_list, @$parallel if (@$parallel && $job_type eq 'children');
for my $dependencies ($chained, $directly_chained, $parallel) {
    next if (@$dependencies && $dependencies == $parallel && $job_type eq 'children');
    clone_job($_, $options, $clone_map, $depth + 1) for @$dependencies;
}
```
That way we remove a bit of indentation there, and I think the call to `push @$child_list, ...` is clearear.

Or am I missing something?

Reply:
Thanks for your help! It works and remove one level of indentation. Added!